pytorch_to_returnn/torch/nn/modules/linear.py

```diff
@@ -123,6 +122,41 @@ def create_returnn_layer_dict(self, *inputs: Tensor, **kwargs) -> Dict[str, Any]
     return {"class": "dot", "red1": red1, "red2": red2, "var1": var1, "var2": var2, "from": sources}
 
 
+class DotProductAlongDim(Module):
+  """
+  Maps to RETURNN DotLayer for a special case
+  """
+  is_original_torch_module = False
+
+  def __init__(self, dim: int):
+    super(DotProductAlongDim, self).__init__()
+    self.dim = dim
+
+  def _get_output_shape_from_returnn(self, inputs_flat: List[Tensor], layer: LayerBase
```

Comment:
Do you need the custom `_get_output_shape_from_returnn` for this layer? I would have assumed that the base logic should work.